src/containers/PostForm.js

```diff
@@ -1,35 +1,72 @@
-import React from 'react'
+import React, { Component, PropTypes } from 'react'
 import { connect } from 'react-redux'
 import { reduxForm } from 'redux-form'
-import { fromForm } from 'store/selectors'
-import { postCreate } from 'store/actions'
+import merge from 'lodash/merge'
+import { fromForm, fromEntities } from 'store/selectors'
+import { postCreate, postRead, postUpdate } from 'store/actions'
 import { createValidator, required } from 'services/validation'
 
 import { PostForm } from 'components'
 
-const PostFormContainer = props => <PostForm {...props} />
+class PostFormContainer extends Component {
+  static propTypes = {
+    id: PropTypes.any,
+    request: PropTypes.func.isRequired
+  }
+
+  componentDidMount() {
+    if (this.props.id) {
+      this.props.request()
+    }
+  }
 
-const onSubmit = (data, dispatch) => new Promise((resolve, reject) => {
-  dispatch(postCreate.request(data, resolve, reject))
+  render() {
+    const props = this.props
+    return <PostForm {...props} />
+  }
+}
+
+const onSubmit = (data, dispatch, state) => new Promise((resolve, reject) => {
+  if (data.id) {
+    return dispatch(postUpdate.request(state.initialValues, data, resolve, reject))
+  }
+  return dispatch(postCreate.request(data, resolve, reject))
 })
 
 const validate = createValidator({
   title: [required],
   body: [required]
 })
 
-const mapStateToProps = (state) => ({
-  initialValues: {
-    _csrf: fromForm.getCsrfToken(state)
+const mapStateToProps = (state, { id }) => {
+  if (id) {
+    return ({
+      initialValues: merge(
+        fromEntities.getDetail(state, 'post', id),
+        {
+          _csrf: fromForm.getCsrfToken(state)
+        }
+      )
+    })
```

Comment:
I wouldn't use `lodash` here, but [`object spread`](https://github.com/sebmarkbage/ecmascript-rest-spread):
```js
return {
  initialValues: {
    _csrf: fromForm.getCsrfToken(state),
    ...fromEntities.getDetail(state, 'post', id)
  }
}
```

Also, you don't need to wrap with `()`